first crack at adapter-auto

Rich-Harris opened the pull request:
#2805. For now this just adds adapter-vercel and adapter-netlify so that we can quickly try it out (difficult to test without publishing the package), but if it works we can immediately add other adapters (though I think for now it's really just Cloudflare Pages? not sure if it makes sense to try and detect that we should use adapter-node or adapter-static)

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

sjroe commented on packages/adapter-auto/package.json:
```diff
@@ -0,0 +1,32 @@
+		"format": "npm run check-format -- --write",
+		"check-format": "prettier --check . --config ../../.prettierrc --ignore-path .gitignore"
+	},
+	"dependencies": {
```
Missing `"@sveltejs/adapter-cloudflare": "workspace:*"`

Rich-Harris replied:
thanks 👍

Rich-Harris commented:
One alternative we might consider is keeping the `test()` logic but then only importing the underlying adapter if it's installed, otherwise prompting the user to install it. This eliminates the risk of `adapter-auto` growing crufty over time, but does introduce some additional friction.

An in-between option would be to install it automatically during the build (e.g. if `process.env.VERCEL` then we automatically install `adapter-vercel`). This adds some complexity though (need to consider different package managers etc), and probably isn't as fast as having it installed initially.

Conduitry commented on documentation/docs/10-adapters.md:
```diff
@@ -2,45 +2,98 @@
-...and traditional platforms:
+- Clear out the build directory
+- Output code that:
+  - Calls `init`
```
Can the `init` being referred to here be clarified?

Similarly, can it be clarified where the `render` that's supposed to be called below comes from?

Or do you think that the current instructions will suffice if someone is simultaneously following along reading an existing adapter?

Rich-Harris replied:
yeah these docs were not all that complete. fleshing them out now

Conduitry commented on documentation/docs/10-adapters.md:
```diff
@@ -2,45 +2,98 @@
 
-- [`adapter-node`](https://github.com/sveltejs/kit/tree/master/packages/adapter-node) — for creating self-contained Node apps
-- [`adapter-static`](https://github.com/sveltejs/kit/tree/master/packages/adapter-static) — for prerendering your entire site as a collection of static files
+If possible, we recommend putting the adapter output under the `build/` directory with any intermediate output placed under `'.svelte-kit/' + adapterName`.
```
I'd probably write this as `.svelte-kit/[adapter-name]` or something similar, rather than as a string operation.

Rich-Harris commented:
The adapter API could definitely use far more comprehensive documentation but a) probably best it doesn't derail this PR (since it's really just moving stuff from one file into another) and b) it should probably wait until the API is more final. i addressed the points above though
